compute_endpoint/tests/unit/test_cli_behavior.py

```diff
@@ -449,6 +449,37 @@ def test_start_ep_display_name_in_config(
     assert conf_dict["display_name"] == display_name
 
 
+@pytest.mark.parametrize(
+    "ha_test_info",
+    [
+        ["ep0", None],
+        ["ep1", False],
+        ["ep2", True],
+    ],
+)
+def test_start_ep_high_assurance_in_config(
+    run_line, mock_command_ensure, make_endpoint_dir, ha_test_info
+):
+    ep_name, is_ha = ha_test_info
+
+    conf = mock_command_ensure.endpoint_config_dir / ep_name / "config.yaml"
+    configure_arg = ""
+    auth_policy = str(uuid.uuid4())
+    sub_id = str(uuid.uuid4())
+    if is_ha is not None:
+        configure_arg = (
+            f" --subscription-id {sub_id} --auth-policy "
+            f"{auth_policy} --high-assurance"
+        )
+    run_line(f"configure {ep_name}{configure_arg}")
+
+    with open(conf) as f:
+        conf_dict = yaml.safe_load(f)
+
+    if is_ha is True:
+        assert conf_dict["high_assurance"] == is_ha
```

Comment:
It's mildly weird to not `assert` something in a test (i.e., the else case).  I assume this is because we don't expect `high_assurance` to exist if not specifically set?  In that case, what about 2 separate tests?

But I'm also not clear on what is supposed to happen in the `False` case?  In that case, we _do_ specify `--high-assurance` on the command line ... but choose not to check that it's in the YAML document?

Reply:
The original intention was only to check the truthy case as that's the only part that's new.  But might as well check all 3 cases, but I left it simple/open for the 'else' path and just assert that ha is either not present or False (to be a bit more flexible in the future about whether to require the setting).

Also redid the parameterize to save a line.